app.py

```diff
@@ -80,6 +102,66 @@ def create_photo(user_id, plant_id):
         return jsonify(photo)
     return jsonify({"error": "Failed to upload"})
 
+@app.route('/api/v1/users/<string:user_id>/plants/<string:plant_id>/journal', methods=['POST'])
+def create_journal_entry(user_id, plant_id):
+    user = mongo.db.users.find_one_or_404({'_id': user_id})
+    plant = mongo.db.plants.find_one_or_404({'_id': plant_id})
+    timestamp = int(datetime.now().timestamp() * 1000)
+    journal_entry = {
+        '_id': str(ObjectId()),
+        'plant_id': plant_id,
+        'entry_type': request.json.get('entryType', ''),
+        'timestamp': timestamp,
+        'info': request.json.get('info', ''),
+    }
+    mongo.db.journal.insert_one(journal_entry)
+    return jsonify(journal_entry)
+
+@app.route('/api/v1/users/<string:user_id>/plants/<string:plant_id>/journal/<string:journal_id>', methods=['GET'])
+def get_journal_entry(user_id, plant_id, journal_id):
+    user = mongo.db.users.find_one_or_404({'_id': user_id})
+    plant = mongo.db.plants.find_one_or_404({'_id': plant_id})
+    journal = mongo.db.journal.find_one_or_404({'_id': journal_id})
+    journal_entry = {
+        'id': journal['_id'],
+        'plantId': plant_id,
+        'entryType': journal['entry_type'],
+        'timestamp': journal['timestamp'],
+        'info': journal['info'],
+    }
+    return jsonify(journal_entry)
+
+@app.route('/api/v1/users/<string:user_id>/plants/<string:plant_id>/journal', methods=['GET'])
+def get_plant_journal(user_id, plant_id):
+    user = mongo.db.users.find_one_or_404({'_id': user_id})
+    plant = mongo.db.plants.find_one_or_404({'_id': plant_id})
+    plant_journal = mongo.db.journal.find({'plant_id': plant_id})
+    format_journal = lambda journal: {
+        'id': journal['_id'],
+        'plantId': plant_id,
+        'entryType': journal['entry_type'],
+        'timestamp': journal['timestamp'],
+        'info': journal['info'],
+    }
```

Comment:
We should break this function out at some point, maybe make it part of a model class to validate incoming requests